barretenberg/cpp/src/barretenberg/goblin/goblin.hpp

```diff
@@ -109,7 +109,7 @@ class Goblin {
         auto ultra_proof = prover.construct_proof();
 
         // Construct and store the merge proof to be recursively verified on the next call to accumulate
-        MergeProver merge_prover{ op_queue };
+        MergeProver merge_prover{ circuit_builder.op_queue };
```

Comment:
We need the prover to be dependent on the current circuit builder instead of taking information from the global op_queue.

Reply:
Not really. Since we only did circuits sequentially, we didn't care about this, since both pointers pointed to the same object.